### PR TITLE
Fix #265 file_eq accepts is_bytes to compare binary

### LINKED ISSUE #265: file_eq allow binary comparisons

### PR DESCRIPTION
**Pull request by robnagler**
@gurhar1133 please approve this and merge before making your changes to file_eq that we discussed Friday

- [x] Require test?
   - yes
- [x] Security implications?
    - no
- [x] UI changes reviewed?
    - n/a
- [x] Backwards compatible?
    - yes
- [x] Require doc changes?
    - yes
- [x] Add [DesignHints](https://github.com/radiasoft/pykern/wiki/DesignHints)?
    - no

### CODE REVIEW

**Comment by robnagler** [comment]
@gurhar1133 ping

File: pykern/pkunit.py [hunk 1]
```diff
@@ -549,7 +548,13 @@ def _message(self):
     """
 
     def _set_expect_and_actual(self):
+        (self._read, self._write) = (
```
Comment (gurhar1133) [inline pykern/pkunit.py:551]: why parens around ` (self._read, self._write) `. Is that what black does or is it a radiasoft style guideline to have parens around a tuple unpacking assignment?

File: tests/pkunit_test.py [hunk 2]
```diff
@@ -58,19 +58,25 @@ def test_empty_work_dir():
...
-    d = pykern.pkunit.empty_work_dir()
-    pykern.pkio.write_text(d.join("file_eq3.txt"), "something")
-    with pykern.pkunit.pkexcept("both exist"):
-        pykern.pkunit.file_eq("file_eq3.txt", actual="something else")
+    pkunit.file_eq("file_eq1.json", actual=a)
+    with pkunit.pkexcept(TypeError):
+        pkunit.file_eq("file_eq2.txt", actual=dict())
+    d = pkunit.empty_work_dir()
+    pkio.write_text(d.join("file_eq3.txt"), "something")
+    with pkunit.pkexcept("both exist"):
```
Comment (gurhar1133) [inline tests/pkunit_test.py:72]: Maybe break into smaller subcalls.

**Review by gurhar1133 — APPROVED** [review]
Looks good, just left a question but approved. 
in `test_file_eq` the `with pkexcept` blocks get pretty dense and that part may benefit from being broken into subcalls

File: pykern/pkunit.py (hunk 1 above)
Comment (robnagler) [reply pykern/pkunit.py]: Uh, cuz I had no idea that this works: `a, b = 1, 2`. Note that the other parens are necessary due to the inline `if`. 

Fixed a8fd141

**Pull request merged**